src/backend/catalog/pg_filesystem.c

```diff
@@ -384,7 +384,7 @@ FileSystemGetNameByOid(Oid	fsysOid)
 
 char *fsys_func_type_to_name(FileSystemFuncType ftype)
 {
-	if(ftype < 0 || ftype >= FSYS_FUNC_TOTALNUM)
+	if (!(ftype >= FSYS_FUNC_CONNECT && ftype < FSYS_FUNC_TOTALNUM))
```

Comment:
It does not seem to need a conversion of !. 
The code below seems to apply either unsigned or signed type, which enum may use.
ftype < FSYS_FUNC_CONNECT || ftype >= FSYS_FUNC_TOTALNUM